src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ShareACasePermissionsHandlerTest.java

```diff
@@ -25,23 +26,92 @@
     private ShareACasePermissionsHandler shareACasePermissionsHandler;
 
     @Mock CcdUpdater ccdUpdater;
+    @Mock ProfessionalUsersRetriever professionalUsersRetriever;
 
     @Mock private Callback<AsylumCase> callback;
     @Mock private CaseDetails<AsylumCase> caseDetails;
     @Mock private AsylumCase asylumCase;
 
+    private String someUserId1 = "someUserId1";
+    private String someUserEmail1 = "someUser1@example.com";
+    private String someUserId2 = "someUserId2";
+    private String someUserEmail2 = "someUser2@example.com";
+
+    private ProfessionalUser professionalUserActive = new ProfessionalUser(
+        someUserId1,
+        "someFirstName",
+        "someLastName",
+        someUserEmail1,
+        newArrayList(),
+        "ACTIVE",
+        "someCode",
+        "someMessage"
+    );
+
+    private ProfessionalUser professionalUserNonActive = new ProfessionalUser(
+        someUserId2,
+        "someFirstName",
+        "someLastName",
+        someUserEmail2,
+        newArrayList(),
+        "NONACTIVE",
+        "someCode",
+        "someMessage"
+    );
+
+    private ProfessionalUsersResponse response = new ProfessionalUsersResponse(newArrayList(professionalUserActive, professionalUserNonActive));
 
     @Before
     public void setUp() throws Exception {
-        shareACasePermissionsHandler = new ShareACasePermissionsHandler(ccdUpdater);
-
         when(callback.getCaseDetails()).thenReturn(caseDetails);
         when(caseDetails.getCaseData()).thenReturn(asylumCase);
         when(callback.getEvent()).thenReturn(Event.SHARE_A_CASE);
+
+        DynamicList dynamicList = new DynamicList(
+            new Value(someUserId1, someUserEmail1),
+            newArrayList()
+        );
+
+        when(asylumCase.read(AsylumCaseFieldDefinition.ORG_LIST_OF_USERS, DynamicList.class)).thenReturn(Optional.of(dynamicList));
+
+        when(professionalUsersRetriever.retrieve()).thenReturn(response);
+
+        shareACasePermissionsHandler = new ShareACasePermissionsHandler(ccdUpdater, professionalUsersRetriever);
+
+        reset(ccdUpdater);
+    }
+
+    @Test
+    public void should_not_invoke_ccd_updater_when_user_is_not_valid() {
+
+        DynamicList dynamicList = new DynamicList(
+            new Value("injectedUserId", "injectedUserEmail@example.com"),
+            newArrayList()
+        );
+
+        when(asylumCase.read(AsylumCaseFieldDefinition.ORG_LIST_OF_USERS, DynamicList.class)).thenReturn(Optional.of(dynamicList));
+
+        shareACasePermissionsHandler.handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback);
+
+        verify(ccdUpdater, times(0)).updatePermissions(callback);
```

Comment:
This is fine, but there is also a `Mockito.verifyZeroInteractions();` you could use here.